src/App.tsx

```diff
@@ -15,48 +18,93 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabet = 'alphabet',
+  Length = 'length',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortField: string,
+  reversed: boolean | string | string[],
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.Alphabet:
+          return good1.localeCompare(good2);
+
+        case SortType.Length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+
+    if (reversed) {
+      preparedGoods.reverse();
+    }
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<string>('');
+  const [reversed, setReversed] = useState<boolean>(false);
```

Comment:
it's redundant to specify the primitives
```suggestion
  const [sortField, setSortField] = useState('');
  const [reversed, setReversed] = useState(false);
```